yarn-project/noir-protocol-circuits/src/crates/rollup-lib/src/base/state_diff_hints.nr

```diff
@@ -0,0 +1,28 @@
+use dep::types::{
+    abis::{
+        membership_witness::NullifierMembershipWitness,
+        nullifier_leaf_preimage::NullifierLeafPreimage,
+    },
+    constants::{
+        MAX_NEW_NULLIFIERS_PER_TX,
+        NOTE_HASH_SUBTREE_SIBLING_PATH_LENGTH,
+        NULLIFIER_SUBTREE_SIBLING_PATH_LENGTH,
+        CONTRACT_SUBTREE_SIBLING_PATH_LENGTH,
+        PUBLIC_DATA_SUBTREE_SIBLING_PATH_LENGTH,
+    },
+};
+
+struct StateDiffHints {
+    nullifier_predecessor_preimages: [NullifierLeafPreimage; MAX_NEW_NULLIFIERS_PER_TX],
+    nullifier_predecessor_membership_witnesses: [NullifierMembershipWitness; MAX_NEW_NULLIFIERS_PER_TX],
+
+    sorted_nullifiers: [Field; MAX_NEW_NULLIFIERS_PER_TX],
+    sorted_nullifier_indexes: [u32; MAX_NEW_NULLIFIERS_PER_TX],
+
+    // For inserting the new subtrees into their respective trees:
+    // Note: the insertion leaf index can be derived from the above snapshots' `next_available_leaf_index` values.
```

Comment:
We don't have the "above" snapshots in this above anymore, they are in a different file now 👀

Reply:
Good catch. Addressed in https://github.com/AztecProtocol/aztec-packages/pull/3919/commits/73b8ee920f4e4bf41a6354f2c8413b4925fada31